tests/unit/stage/test_cache.py

```diff
@@ -204,3 +206,19 @@ def _mode(path):
     assert _mode(parent_cache_dir) == dir_mode
     assert _mode(cache_dir) == dir_mode
     assert _mode(cache_file) == file_mode
+
+
+def test_always_changed(mocker):
+    from dvc.repo import Repo
+    from dvc.stage import Stage
+    from dvc.stage.cache import RunCacheNotFoundError, StageCache
+
+    repo = mocker.Mock(spec=Repo)
+    cache = StageCache(repo)
+    stage = Stage(repo, always_changed=True)
+    get_stage_hash = mocker.patch("dvc.stage.cache._get_stage_hash")
```

Comment:
How about we assert that `cache.save` returns None, and no entry exists in `runs` directory?

Reply:
Sorry for not noticing earlier, that it was a "private" API.

Reply:
Ah, okay, there's no `tmp_dir`.

Reply:
@skshetry With StageMock it was not necessary because anything would've got an attribute error if tried to do anything with it, but with a real stage it indeed makes sense as an additional sanity check.